src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -3519,6 +3521,44 @@ internal static bool SerializePeToStream(
         /// </summary>
         internal bool CatchAnalyzerExceptions => Feature("debug-analyzers") == null;
 
+        internal int? DataSectionStringLiteralThreshold
+        {
+            get
+            {
+                if (!_lazyDataSectionStringLiteralThreshold.HasValue)
+                {
+                    _lazyDataSectionStringLiteralThreshold = compute();
```

Comment:
>_lazyDataSectionStringLiteralThreshold = compute(); [](http://example.com/codeflow?start=20&length=51)

```Optional<T>``` is a struct, therefore, this assignment is not guaranteed to be atomic. Therefore, the code is not thread safe.

Reply:
I wasn't sure how to make this thread safe, so I changed it to Lazy, but let me know if you think there is a better way.